src/java.base/share/classes/java/lang/Module.java

```diff
@@ -597,6 +597,15 @@ public boolean isExported(String pn, Module other) {
      *
      * <p> This method does not check if the given module reads this module. </p>
      *
+     * <p> A package {@code p} opened to module {@code M} means that code in
+     * {@code M} can do deep reflection on all types in the package and all
+     * their members for example via
+     * {@link java.lang.reflect.AccessibleObject#setAccessible(boolean) setAccessible},
+     * and if {@code M} reads this module, it can also obtain a
+     * {@link java.lang.invoke.MethodHandles.Lookup Lookup} object that can be used to
+     * {@link java.lang.invoke.MethodHandles.Lookup#defineClass(byte[]) inject classes}
+     * in package {@code p}. </p>
```

Comment:
```suggestion
     * <p> A package {@code p} opened to module {@code M} means that code in
     * {@code M} is allowed to do deep reflection on all types in the package.
     * Further, if {@code M} reads this module it can obtain a
     * {@link java.lang.invoke.MethodHandles.Lookup Lookup} object that is allowed to
     * {@link java.lang.invoke.MethodHandles.Lookup#defineClass(byte[]) define classes}
     * in package {@code p}. </p>
```
Trying to reuse existing terms. I am presuming deep reflection implies on all members and setAccessible so there is no need to mention it?

Also i don't see "inject" used in existing text, so just reuse "define"?

Reply:
Using "define" is fine too. 

This method links to `AccessibleObject.setAccessible` and `MethodHandles.privateLookupIn`.   It may be adequate and no need to mention more.